events/wc_void_events.txt

```diff
@@ -2371,3 +2371,147 @@ narrative_event = {
 		clr_character_flag = found_cthraxxi_bad_end_flag
 	}
 }
+
+### Tekahn's Bargain ###
+# An agent of the Twilight's Hammer visits the Neferset and offers to lift the Curse of Flesh
+narrative_event = {
+	id = WCVOD.10100
+	title = EVTTITLE_WCVOD_10100
+	desc = EVTDESC_WCVOD_10100
+	picture = GFX_evt_cthun_priestess
+	border = GFX_event_narrative_frame_intrigue
+
+	is_triggered_only = yes
+
+	trigger = {
+		has_landed_title = c_neferset	# Must hold the city of Neferset
+		culture = tolvir	# Must be of tol'vir culture
+		is_shadow_priest_class_trigger = yes	# Only shadow priests can be contacted by the entity
+		has_global_flag = uldum_has_opened_flag		# Uldum must not be isolated
+		NOT = { has_global_flag = dark_pharaohs_bargain_flag }	# This is here so event can't fire twice
+	}
+
+	immediate = {
+		set_global_flag = dark_pharaohs_bargain_flag
+		save_event_target_as = target_dark_pharaoh
+	}
+
+	option = {
+		name = EVTOPTA_WCVOD_10100
+
+		event_target:target_dark_pharaoh = { narrative_event = { id = WCVOD.10105 } }
+		event_target:target_dark_pharaoh = { give_nickname = nick_the_dark_pharaoh }
+
+		ai_chance = {
+			factor = 1
+		}
+	}
+
+	option = {
+		name = EVTOPTB_WCVOD_10100
+
+		custom_tooltip = {
+			text = you_reject_the_offer
+		}
+	}
+}
+
+# A ritual commences, 'Tekahn' regains his stone body
+narrative_event = {
+	id = WCVOD.10105
+	title = EVTTITLE_WCVOD_10105
+	desc = EVTDESC_WCVOD_10105
+	picture = GFX_evt_cthun_priestess
+	border = GFX_event_narrative_frame_intrigue
+
+	is_triggered_only = yes
+
+	# I shall serve the Old Gods
+	option = {
+		name = EVTOPTA_WCVOD_10105
+
+		become_stone_tolvir_old_gods_effect = yes
+	}
```

Comment:
What if you already do?

Reply:
added a requirement for the opening event that you have to be independent of Old God influence at the time.